infra/blueprint-test/test/terraform_simple_base_api_module_test.go

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/tft"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSimpleTFModule(t *testing.T) {
+	nt := tft.Init(t,
+		tft.WithTFDir("../examples/simple_tf_module"),
+		tft.WithSetupPath("setup/simple_tf_module"),
```

Comment:
We should make the example structure for tests mirror what actual modules look like:

```
- test/
-- modules/
--- simple_tf_module
----- test/
------- setup/
--- other_tf_module
----- test/
------- setup/
```

Reply:
I am actually planning on using top level `examples` to [showcase real examples](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/tree/add-tf-bptest/infra/blueprint-test/examples). This will increase discoverability for common test patterns. In `test/*` I plan on keeping some single e2e tests exercising API compatibility and finally along side each `pkg` I will add unit tests for that package.